released_plugins/v3d_plugins/MIND/soma_segmentation_plugin.cpp

```diff
@@ -264,5 +278,15 @@ void reconstruction_func(V3DPluginCallback2 &callback, QWidget *parent,
 
   callback.openROI3DWindow(curwin);
 
+  // Update landmark and set z thickness
+  View3DControl *v3dlocalcontrol = callback.getLocalView3DControl(curwin);
+  if (v3dlocalcontrol) {
+    v3dlocalcontrol->updateLandmark();
```

Comment:
Interesting - so there is a built-in function that allows you to control the window, change its thickness, and update landmarks? And the left behind landmark is a result of that function, not us?

Reply:
Yeah there is a built-in function that controls the 3D view. It does essentially the same as clicking the buttons on the GUI control pane, but in code format.

What do you mean by
> And the left behind landmark is a result of that function, not us?

Reply:
Awesome. Sorry I mean that the solution you showed had a problem where the updated landmarks were there, but so were the old landmarks. That is a result of a problem in vaa3D's implementation, no?

Reply:
It's not that the old landmarks are there, it's that all landmarks of the image show up, even if they are not in the ROI. But yes, I think it's something from Vaa3D's implementation.

Reply:
Oh I see! Ok that's not really a problem then! So when you change the z scale, the landmarks change accordingly? We are good then with this :)